internal/common/mqtt/mqttconfig.go

```diff
@@ -92,21 +107,36 @@ func addClientData(client *Client, clientID string) {
 }
 
 //SetBrokerURL returns the broker url for connection
-func (c *Client) SetBrokerURL(protocol string) string {
-	return fmt.Sprintf("%s://%s:%d", protocol, c.Host, c.Port)
+func (c *Client) SetBrokerURL() string {
+	return fmt.Sprintf("%s://%s:%d", c.protocol, c.Host, c.Port)
 }
 
-func checkforConnection(brokerURL string, mqttClient *Client) int {
+func checkforConnection(brokerURL string, mqttClient *Client, mqttPort uint) int {
 	if mqttClient == nil {
 		return 0
 	}
 	log.Info(logPrefix, mqttClient.URL)
-	connURL := fmt.Sprintf("%s://%s:%d", "tcp", brokerURL, mqttPort)
+	connURL := fmt.Sprintf("%s://%s:%d", mqttClient.protocol, brokerURL, mqttPort)
 	return strings.Compare(connURL, mqttClient.URL)
 }
 
+//NewTLSConfig creates a tls config for mqtt client
+func NewTLSConfig() *tls.Config {
```

Comment:
Please add a return for an error.
```suggestion
func NewTLSConfig() (*tls.Config, error) {
```

Reply:
Done!!